app/lib/actions/katello/host/remove_subscriptions.rb

```diff
@@ -0,0 +1,29 @@
+module Actions
+  module Katello
+    module Host
+      class RemoveSubscriptions < Actions::Base
+        middleware.use Actions::Middleware::KeepCurrentUser
+
+        def plan(host, entitlements)
+          entitlements.each do |entitlement|
+            plan_action(::Actions::Candlepin::Consumer::RemoveSubscription, :uuid => host.subscription_facet.uuid,
+                        :entitlement_id => entitlement['id'])
+            plan_self(:host_name => host.name)
+          end
+        end
+
+        def rescue_strategy
+          Dynflow::Action::Rescue::Skip
+        end
+
+        def finalize
+          ::Katello::Pool.import_all
```

Comment:
adding and removing a subscription can cause other pools to be created or deleted, right @thomasmckay  ?